exercises/collatz-conjecture/test/Tests.hs

```diff
@@ -0,0 +1,60 @@
+{-# OPTIONS_GHC -fno-warn-type-defaults #-}
+{-# LANGUAGE RecordWildCards #-}
+
+import Data.Foldable     (for_)
+import Test.Hspec        (Spec, describe, it, shouldBe)
+import Test.Hspec.Runner (configFastFail, defaultConfig, hspecWith)
+
+import CollatzConjecture (collatz)
+
+main :: IO ()
+main = hspecWith defaultConfig {configFastFail = True} specs
+
+specs :: Spec
+specs = describe "collatz" $ for_ cases test
+  where
+
+    test Case{..} = it description assertion
+      where
+        assertion = collatz number `shouldBe` fromIntegral expected
+
+
+data Case = Case { description :: String
+                 , number      :: Integer
+                 , expected    :: Integer
+                 }
+
+cases :: [Case]
+cases = [ Case { description = "collatz 1 don't require any steps."
+               , number      = 1
+               , expected    = 0
+               }
+        , Case { description = "collatz 2 require 1 step."
+               , number      = 2
```

Comment:
on my advice, you did change the problem version to 1.1.1.1, to match the 1.1.1 of https://github.com/exercism/problem-specifications/blob/master/exercises/collatz-conjecture/canonical-data.json, but you did not then use the test cases in https://github.com/exercism/problem-specifications/blob/master/exercises/collatz-conjecture/canonical-data.json. Can you do one of the two:

1. Use the test cases that appear in https://github.com/exercism/problem-specifications/blob/master/exercises/collatz-conjecture/canonical-data.json
2. Confirm that the test cases in https://github.com/exercism/problem-specifications/blob/master/exercises/collatz-conjecture/canonical-data.json are unsuitable for the Haskell track, and thus we cannot use that file, and thus we must change our version to 0.1.0.1

Reply:
so it's ok if i change it from 1.1.1.1 to 1.1.1 and use [this cases](https://github.com/exercism/problem-specifications/blob/master/exercises/collatz-conjecture/canonical-data.json) ?

Reply:
1.1.1 is not a valid Haskell track version. The two choices are:

1. Use 1.1.1.1 and the cases in https://github.com/exercism/problem-specifications/blob/master/exercises/collatz-conjecture/canonical-data.json
2. Use 0.1.0.1 and whatever cases desired, with an explanation of why https://github.com/exercism/problem-specifications/blob/master/exercises/collatz-conjecture/canonical-data.json cannot be used.

Reply:
yeah, i forgot the extra 1 again. sorry about that. i'll use the first one